v2/pkg/engine/resolve/fetchtree_test.go

```diff
@@ -0,0 +1,142 @@
+package resolve
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/wundergraph/graphql-go-tools/v2/pkg/ast"
+)
+
+func TestFetchTreeQueryPlanNode_PrettyPrint_Trigger(t *testing.T) {
+	t.Run("just a trigger", func(t *testing.T) {
+		fetches := Sequence()
+		fetches.Trigger = &FetchTreeNode{
+			Kind: FetchTreeNodeKindTrigger,
+			Item: &FetchItem{
+				Fetch: &SingleFetch{
+					FetchDependencies: FetchDependencies{
+						FetchID: 0,
+					},
+					Info: &FetchInfo{
+						DataSourceID:   "0",
+						DataSourceName: "country",
+						QueryPlan: &QueryPlan{
+							Query: `subscription {
+    countryUpdated {
+        name
+    }
+}`,
+						},
+					},
+				},
+				ResponsePath: "countryUpdated",
+			},
+		}
+
+		queryPlan := fetches.QueryPlan()
+		actual := queryPlan.PrettyPrint()
+
+		expected := `
+QueryPlan {
+  Subscription {
+    Primary: {
+      Fetch(service: "country") {
+        {
+            countryUpdated {
+                name
+            }
+        }
+      }
+    },
+  }
+}`
+
+		assert.Equal(t, strings.TrimSpace(expected), strings.TrimSpace(actual))
+	})
+	t.Run("trigger with representation call", func(t *testing.T) {
+		fetches := Sequence()
+		fetches.Trigger = &FetchTreeNode{
+			Kind: FetchTreeNodeKindTrigger,
+			Item: &FetchItem{
+				Fetch: &SingleFetch{
+					FetchDependencies: FetchDependencies{
+						FetchID: 0,
+					},
+					Info: &FetchInfo{
+						DataSourceID:   "0",
+						DataSourceName: "country",
+						QueryPlan: &QueryPlan{
+							Query: `subscription {
+    countryUpdated {
+        name
+        time {
+            local
+        }
+    }
+}`,
+						},
+					},
+				},
+				ResponsePath: "countryUpdated",
+			},
+		}
+		fetches.ChildNodes = []*FetchTreeNode{{
+			Kind: FetchTreeNodeKindSingle,
+			Item: &FetchItem{
+				Fetch: &SingleFetch{
+					FetchDependencies: FetchDependencies{
+						FetchID:           1,
+						DependsOnFetchIDs: []int{0},
+					},
+					Info: &FetchInfo{
+						DataSourceID:   "1",
+						DataSourceName: "time",
+						OperationType:  ast.OperationTypeQuery,
+						QueryPlan: &QueryPlan{
+							Query: "query($representations: [_Any!]!){\n    _entities(representations: $representations){\n        ... on Time {\n            __typename\n            local\n        }\n    }\n}",
+						},
+					},
+				},
+				ResponsePath: "countryUpdated.time",
+			},
+		}}
+
+		queryPlan := fetches.QueryPlan()
+		actual := queryPlan.PrettyPrint()
+
+		expected := `
+QueryPlan {
+  Subscription {
+    Primary: {
+      Fetch(service: "country") {
+        {
+            countryUpdated {
```

Comment:
Overall looks good, I'm just wondering if we should write this as `subscription countryUpdated`.
Just having the root field could be interpreted as a query.
I was also thinking if it would make sense to prefix mutations with `mutation`.
Query can be just a selection set because it's kind of well known that just having a selection set without query, mutation or subscription is understood as a query.

Reply:
In this case, surrounding block is `Subscription {}`, it can be concluded that it is the subscription and not the query. But for other cases like mutation vs query it is not conclusive just from the query plan. Maybe @devsergiy has some ideas how we can streamline query plans.